src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -64,21 +74,65 @@ public Context CreateContext(IFeatureCollection contextFeatures)
                 // Non-inline
                 LogRequestStarting(httpContext);
             }
-
             if (diagnoticsEnabled)
```

Comment:
Besides sampling, we want to support other scenarios. 
Note that Activity carries duration, that makes Begin event almost redundant and increases log verbosity.
So there are 2 scenarios:
1. Users may listen to both: Begin and End Request
2. Users may want to receive End only.

Currently `diagnoticsEnabled` depends on Begin and changing it might break existing customers that listen to Begin only. I'm not sure if it makes any sense, since End event carries response.

I suggest 2 options:
1. Change diagnoticsEnabled to `_diagnosticSource.IsEnabled(DiagnosticsEndRequestKey)` and most probably no one will be hurt.

2. Deprecate Begin/End events and add a set of Activity events: "Microsoft.AspNetCore.Activity" (for IsEnabled), and Microsoft.AspNetCore.Activity.Start/Stop. This will not break anyone, but would require some deprecation plan and won't bring Activity to the old consumers

If we leave it as is, End-only scenario will be impossible. It also will be inconsistent with System.Net.Http instrumentation approach and ASP.NET Classic approach that is being implemented